docs/bulk-retries-design.md

```diff
@@ -18,22 +18,33 @@ When ServiceControl restarts, the in-memory representations are rebuilt by aggre
 
 When a retry batch is first created it has this state. This means that ServiceControl is finding and marking failed messages as belonging to this batch.
 
-To do this, a new document is created for each failed message. Each one has an id `FailedMessageRetry/{messageId}` and contains the failed message id and the retry batch id. As only one document with this key can exist at a time, this ensures that a Failed Message can only ever belong to a single batch. This document will exist until such a time as a new Failed Message with same Id comes through the error queue. This guarantees that there can only be one outstanding retry for a failed message at a time.
+To do this, a new document is created for each failed message. Each one has an id `FailedMessageRetry/{messageId}` and contains the failed message id and the retry batch id. As only one document with this key can exist at a time, this ensures that a Failed Message can only ever belong to a single batch. This FailedMessageRetry document will exist until such a time as a new Failed Message with same Id comes through the error queue. This guarantees that there can only be one outstanding retry for a failed message at a time.
 
 When all messages have been marked, a list of the `FailedMessageRetry` ids is appended to the batch and the batch changes status to `Staging`. The list inside of the Batch may contain failed messages which do not belong to this batch (because another batch claimed them in parallel). These will get filtered out during staging (below).
 
-When ServiceControl starts up it will attempt to adopt any batches that it finds in this status and move them to `Staging`. This can only happen if the SC process stops during the above process. Any documents that were already marked will be added to the batch. Any documents that had not yet been marked are ignored and will have to be retried again by the user. When SC starts up, it will generate a `Session ID` GUID. This GUID is stamped onto each new batch as it is created. This is how SC can tell if a batch is from a previous session and adopt it. Only Batches with a non-current session Id will be adotped by the orphan batch process.
+When ServiceControl starts up it will attempt to adopt (what does adopt mean?) any batches that it finds in this status and move them to `Staging`. This can only happen if the ServiceControl process stops during the above process. Any documents that were already marked will be added to the batch. Any documents that had not yet been marked are ignored and will have to be retried again by the user. (<- what does this mean?) When ServiceControl starts up, it will generate a `Session ID` GUID. This GUID is stamped onto each new batch as it is created. This is how ServiceControl can tell if a batch is from a previous session and adopt it. Only Batches with a non-current session Id will be adopted by the orphan batch process.
```

Comment:
```suggestion
When ServiceControl starts up it will attempt to adopt any batches that it finds in this status and move them to `Staging`. This can only happen if the ServiceControl process stops during the above process. Any documents that were already marked will be added to the batch. Any documents that had not yet been marked are ignored and will have to be retried again by the user.

When a bulk retry is issued a background process is started that looks for all of the messages that match its criteria and assign them to a retry batch id (in lots of 1,000). If the process is killed before it is done, the details of what you were trying to retry are gone. Any documents that had already marked for retry will be retried. Any that were not marked for retry will require manual intervention to retry

When ServiceControl starts up, it will generate a `Session ID` GUID. This GUID is stamped onto each new batch as it is created. This is how ServiceControl can tell if a batch is from a previous session and adopt it. Only Batches with a non-current session Id will be adopted by the orphan batch process.
```